src/vcpkg/base/downloads.cpp

```diff
@@ -579,6 +582,45 @@ namespace vcpkg
         return res;
     }
 
+    std::string format_url_query(const std::string& base_url, const std::vector<std::string>& query_params)
+    {
+        if (query_params.empty())
+        {
+            return base_url;
+        }
+
+        std::string query = Strings::join("&", query_params);
+
+        return base_url + "?" + query;
+    }
+
+    ExpectedL<std::string> invoke_http_request(const std::string& method,
+                                               const View<std::string>& headers,
+                                               const std::string& url,
+                                               const std::string& data)
+    {
+        Command cmd;
+        cmd.string_arg("curl").string_arg("-s").string_arg("-L");
+        cmd.string_arg("-H").string_arg(
+            fmt::format("User-Agent: vcpkg/{}-{} (curl)", VCPKG_BASE_VERSION_AS_STRING, VCPKG_VERSION_AS_STRING));
```

Comment:
Should we add a user agent to other `downloads.cpp` things?

Reply:
I don't believe so. I added a user agent here because Michael Price recommended it in case we ever need help debugging a GHA thing.